File: 1281. Subtract the Product and Sum of Digits of an Integer/solution.cpp
#include <iostream>

int subtractProductAndSum(int n) {
  int mult = 1, add = 0;
  while (n != 0) {
    int digit = n % 10;
    mult *= digit;
    add += digit;
    n /= 10;
  }
  return mult - add;
}

int main() {
  int num = 234;
  std::cout << subtractProductAndSum(num);
}

/*
234 = 15
4421 = 21
*/

File: 1281. Subtract the Product and Sum of Digits of an Integer/solution.js
var subtractProductAndSum = function(n) {
  const arr = n.toString().split('');
  const add = arr.reduce((a, b) => Number(a) + Number(b));
  const mult = arr.reduce((a, b) => Number(a) * Number(b));
  // Instead of two .reduce() calls, we could use one for loop, updating both add and mult with each pass
  return mult - add;
};

const num = 234;
console.log(subtractProductAndSum(num));

/*
234 = 15
4421 = 21
*/